File: app/src/main/java/com/example/audiobookmark/MainActivity.java
package com.example.audiobookmark;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class MainActivity extends AppCompatActivity {
    private BookmarkViewModel viewModel;
    static final String TAG = "MainActivity";


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toolbar myToolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(myToolbar);

        viewModel = ViewModelProviders.of(this).get(BookmarkViewModel.class);


        RecyclerView recyclerView = findViewById(R.id.recyclerview);
        final BookmarkListAdapter adapter = new BookmarkListAdapter(this);
        recyclerView.setAdapter(adapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        recyclerView.addItemDecoration(new DividerItemDecoration(recyclerView.getContext(), DividerItemDecoration.VERTICAL));

        viewModel.getAllBookmarks().observe(this, new Observer<List<Bookmark>>() {
            @Override
            public void onChanged(List<Bookmark> words) {
                adapter.setWords(words);
            }
        });

        Intent intent = new Intent(this, MediaCallbackService.class)
                .setAction(MediaCallbackService.START_ACTION);
        startService(intent);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.action_menu, menu);
        return true;
    }


    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            // action with ID action_refresh was selected
            case R.id.action_settings:
                startActivity(new Intent(this, SettingsActivity.class));
                /*
                Example how to give task to service:
                Intent intent = new Intent(this, MediaCallbackService.class)
                        .setAction(MediaCallbackService.PLAY_ACTION);
                startService(intent);*/
                break;
            default:
                break;
        }

        return true;
    }
}

File: app/src/main/java/de/augmentedmind/stopit/ui/MainActivity.kt
package de.augmentedmind.stopit.ui

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import de.augmentedmind.stopit.*
import de.augmentedmind.stopit.service.MediaCallbackService

class MainActivity : AppCompatActivity() {
    private lateinit var viewModel: BookmarkViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val myToolbar = findViewById<View>(R.id.toolbar) as Toolbar
        setSupportActionBar(myToolbar)
        viewModel = ViewModelProvider(this).get(BookmarkViewModel::class.java)
        val recyclerView = findViewById<RecyclerView>(R.id.recyclerview)
        val adapter = BookmarkListAdapter(this)
        recyclerView.adapter = adapter
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.addItemDecoration(DividerItemDecoration(recyclerView.context,
                DividerItemDecoration.VERTICAL))
        viewModel.allBookmarks.observe(this, Observer { bookmarks ->
            bookmarks?.let { adapter.setBookmarks(bookmarks) }
        })
        val intent = Intent(this, MediaCallbackService::class.java)
                .setAction(MediaCallbackService.START_ACTION)
        startService(intent)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.action_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.action_settings -> startActivity(Intent(this, SettingsActivity::class.java))
            else -> {
            }
        }
        return true
    }

    companion object {
        const val TAG = "MainActivity"
    }
}